apps/gateway/lib/gateway/controllers/arena/currency_controller.ex

```diff
@@ -22,7 +22,7 @@ defmodule Gateway.Curse.Controllers.Users.CurrencyController do
            {:add_currency, Currencies.add_currency_by_name_and_game(user_id, currencty_name, user.game_id, amount)} do
       send_resp(conn, 200, Jason.encode!(Map.take(user_currency, [:amount, :user_id, :currency_id])))
     else
-      {:get_user, _} -> send_resp(conn, 400, "User not found")
+      {:get_user, _} -> send_resp(conn, 404, "User not found")
       {:curse_user, _} -> send_resp(conn, 400, "User from another game")
       {:add_currency, _} -> send_resp(conn, 400, "Couldn't add currency")
```

Comment:
```suggestion
      {:add_currency, _} -> send_resp(conn, 404, "Currency not found")
```

This is the only way this path can be reached! Makes for a clearer error :P

Reply:
we can also have an error in the  user currency changeset, so i would let this to be more general

Reply:
Right but now we cannot have an error in the UserCurrency changeset though, there's no other possible error because we're capturing the non-existing user beforehand. If you're afraid this might change in the future, that's valid, but I think it'd be a nice addition

Reply:
Let's change it here and iterate if we ever validate currencies then